From 5f91a2cfbdab35f62c58553f75588151f42d5cff Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 13 Aug 2024 17:39:46 +0200
Subject: [PATCH 660/954] fix(stm32mp2): stm32_otp_is_closed_device should
 return false

If Cortex-M33 is TDCID we cannot know the status of BSEC. The function
stm32mp_check_closed_device() can then not rely on this to know if
the device is closed. We can only rely on lifecycle OTP. Change the
return of stm32_otp_is_closed_device() used when M33 is TDCID to
false, to be able to check lifecycle OTP value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b95f352edc7f0644649ccdb20a8c8834816fee1b fix(stm32mp2): stm32_otp_is_closed_device should return false]
Change-Id: Ia16a5744c3065fed5039e630cde725fd88d719ce
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/398289
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_otp.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_otp.c b/plat/st/stm32mp2/stm32mp2_otp.c
index 757bb7dc61..b6ee09d367 100644
--- a/plat/st/stm32mp2/stm32mp2_otp.c
+++ b/plat/st/stm32mp2/stm32mp2_otp.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2023-2024, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -38,5 +38,9 @@ uint32_t stm32_otp_read_sw_lock(uint32_t otp, bool *value)
 
 bool stm32_otp_is_closed_device(void)
 {
-	return false;
+	/*
+	 * As we cannot check BSEC status, return true here and only rely on lifecycle OTP
+	 * to check the device is closed in stm32mp_check_closed_device().
+	 */
+	return true;
 }
-- 
2.53.0

